sources/oauth: add WeChat type

Anduin2017 opened the pull request:
<!--
👋 Hi there! Welcome.

Please check the Contributing guidelines: https://docs.goauthentik.io/docs/developer-docs/#how-can-i-contribute
-->

## Details

<!--
Explain what this PR changes, what the rationale behind the change is, if any new requirements are introduced or any breaking changes caused by this PR.

Ideally also link an Issue for context that this PR will close using `closes #`
-->
This pull request introduces support for WeChat (Weixin) as an OAuth 2.0 source. This allows administrators to configure WeChat as a login provider, enabling users to authenticate using the "WeChat for Websites" (PC QR code scanning) flow.

---

## Checklist

-   [x] Local tests pass (`ak test authentik/`)
-   [x] The code has been formatted (`make lint-fix`)

If an API change has been made

-   [x] The API schema has been updated (`make gen-build`)

If changes to the frontend have been made

-   [x] The code has been formatted (`make web`)

If applicable

-   [x] The documentation has been updated
-   [x] The documentation has been formatted (`make docs`)

Anduin2017 committed:
Add wechat.

Anduin2017 committed:
Refactor comments and formatting in wechat.py

Signed-off-by: Anduin Xue <anduin@aiursoft.com>

Anduin2017 commented:
@rissson Can you trigger the workflow? I need to make sure the lint pass.

Anduin2017 committed:
Fix lint.

Signed-off-by: Anduin Xue <anduin@aiursoft.com>

Anduin2017 committed:
Fix lint.

Anduin2017 committed:
fix: Rename `WeChat` enum member to `Wechat` for consistency

Anduin2017 committed:
docs: Add WeChat social login integration guide.

Anduin2017 commented:
Hi, @rissson !

All issues I can fixed are fixed.

However, I'm getting:

```log
=========================== short test summary info ============================
FAILED tests/e2e/test_provider_saml.py::TestProviderSAML::test_sp_initiated_implicit_post_buffer - django.core.management.base.CommandError: Database test_authentik couldn't be flushed. Possible reasons:
  * The database isn't running or isn't configured correctly.
  * At least one of the expected database tables doesn't exist.
  * The SQL was invalid.
Hint: Look at the output of 'django-admin sqlflush'. That's the SQL this command wasn't able to run.
============= 1 failed, 11 passed, 1 warning in 300.59s (0:05:00) ==============
```

In https://github.com/goauthentik/authentik/actions/runs/19713888596/job/56482250350?pr=18086

<img width="2560" height="1202" alt="image" src="https://github.com/user-attachments/assets/32ae4413-8947-423c-9fe0-230dce72ab00" />

This error seems not related to my change. Can you help confirm and engage related team for review? Thanks a lot!!

Anduin2017 commented:
@GirlBossRush @tanberry  Hi Authentik Dev team!

Our company need this feature to be landed soon. It would be great if it can be released before 2026. Can  you help me taking a look?

fheisler commented:
Hi @Anduin2017, thank you for the carefully prepared contribution!

This was originally presented as a company needing a new feature within a deadline, which is a large part of what we support for paying enterprise customers, regardless of who writes the code:

> Our company need this feature to be landed soon. It would be great if it can be released before 2026.

However, it sounds like this is not the case:

> We do not need this PR to be merged to use it. I took the extra time [...] purely out of the open-source spirit to give this major feature back to the community.

That's great! As such, this PR is going through our usual review process as @rissson mentioned.

We're already preparing for a `2025.12` release and reviewing a number of other large contributions for inclusion, so I can't guarantee that it will be included in December, but it looks like we've already gotten backend approval and just need frontend + docs reviews.

dewi-tik committed:
Docs updates

dewi-tik commented on website/docs/users-sources/sources/social-logins/wechat/index.md:
```diff
@@ -0,0 +1,70 @@
+To integrate WeChat with authentik you will need to register a "Website Application" (网站应用) on the [WeChat Open Platform](https://open.weixin.qq.com/).
+
+1. Register for a developer account on the [WeChat Open Platform](https://open.weixin.qq.com/).
+2. Create a new "Website Application".
```
There needs to be more guidance on how to create a "Website Application". For example where in the WeChat Open Platform UI is this done?

Anduin2017 committed:
Revise WeChat integration instructions

Updated instructions for creating a WeChat Website Application and added details about scopes and user attribute mappings.

Signed-off-by: Anduin Xue <anduin@aiursoft.com>

dewi-tik committed:
Prettier

dewi-tik reviewed (approved):
Docs approval

Anduin2017 committed:
Update wechat.py

Signed-off-by: Anduin Xue <anduin@aiursoft.com>

rissson committed:
Merge branch 'main' into patch-3

tanberry reviewed (approved):
Thanks @Anduin2017 !
